File: DAMOS/trunk/tests/org.eclipselabs.damos.codegen.c.tests/gtests/MatrixMultiplyTest.cpp
#include "gtest/gtest.h"
#include "MatrixMultiplyTest.h"

TEST(MatrixMultiplyTest, MatrixMultiply) {
	Input input;
	Output output;

	initialize();

	input.inA = 2;
	input.inB = 15;

	execute(&input, &output);

	EXPECT_DOUBLE_EQ(94.0, output.out[0][0]);
	EXPECT_DOUBLE_EQ(100.0, output.out[0][1]);
	EXPECT_DOUBLE_EQ(230.5, output.out[1][0]);
	EXPECT_DOUBLE_EQ(245.6, output.out[1][1]);
	EXPECT_DOUBLE_EQ(364.0, output.out[2][0]);
	EXPECT_DOUBLE_EQ(388.0, output.out[2][1]);
	EXPECT_DOUBLE_EQ(499.0, output.out[3][0]);
	EXPECT_DOUBLE_EQ(532.0, output.out[3][1]);
}

File: DAMOS/trunk/tests/org.eclipselabs.damos.codegen.c.tests/src/org/eclipse/damos/codegen/c/gtests/MatrixMultiplyTest.java
/****************************************************************************
 * Copyright (c) 2008, 2012 Andreas Unger and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *    Andreas Unger - initial API and implementation 
 ****************************************************************************/

package org.eclipse.damos.codegen.c.gtests;

import org.eclipse.damos.codegen.c.test.AbstractGeneratorGTest;
import org.eclipse.damos.codegen.c.test.GTest;
import org.eclipse.damos.dml.Block;
import org.eclipse.damos.dml.Inport;
import org.eclipse.damos.dml.Outport;
import org.junit.Before;

/**
 * @author Andreas Unger
 *
 */
@GTest(sourceFile="gtests/MatrixMultiplyTest.cpp", program="gtests/MatrixMultiplyTest")
public class MatrixMultiplyTest extends AbstractGeneratorGTest {

	@Before
	@Override
	public void setUp() {
		super.setUp();
		
		Inport inA = createInport("InA", createIntegerTypeSpecification(), 1);
		Inport inB = createInport("InB", createIntegerTypeSpecification(), 1);
		Block structConstructionTest = createTestBlock("MatrixMultiplyTest", "MatrixMultiplyTest");
		Outport outport = createOutport("Out", createDataTypeSpecification("real[4, 2]"));
		
		connect(inA, structConstructionTest, 0);
		connect(inB, structConstructionTest, 1);
		connect(structConstructionTest, outport);

		generateAndCompile();
	}

}
